.github/workflows/gradle.yaml

```
@@ -24,18 +24,18 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
-      - name: Set up JDK 17
+      - name: Set up JDK 21
         uses: actions/setup-java@v3
         with:
-          java-version: '17'
+          java-version: '21'
           distribution: 'temurin'
       - name: Validate Gradle wrapper
-        uses: gradle/wrapper-validation-action@v1
+        uses: gradle/actions/wrapper-validation@v3
       - name: Build with Gradle
         uses: gradle/gradle-build-action@v2
         with:
           arguments: build
```

> As of v3 this action has been superceded by gradle/actions/wrapper-validation. Any workflow that uses gradle/wrapper-validation-action@v3 will transparently delegate to gradle/actions/wrapper-validation@v3.

https://github.com/gradle/wrapper-validation-action

> As of v3 this action has been superceded by gradle/actions/setup-gradle. Any workflow that uses gradle/gradle-build-action@v3 will transparently delegate to gradle/actions/setup-gradle@v3.

https://github.com/gradle/gradle-build-action

```suggestion
        uses: gradle/actions/wrapper-validation@v3
      - name: Build with gradle
        run: ./gradlew build
      - uses: actions/upload-artifact@v4
```